huaweicloud/services/evs/data_source_huaweicloud_evs_volumes.go

```diff
@@ -0,0 +1,242 @@
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"shareable": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"server_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"volumes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"attachments": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"attached_at": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"attached_mode": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"device_name": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"server_id": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+								},
+							},
+						},
+						"availability_zone": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"bootable": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"service_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"shareable": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+						"size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"create_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"update_at": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"tags": {
+							Type:     schema.TypeMap,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"wwn": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func buildQueryOpts(d *schema.ResourceData) cloudvolumes.ListOpts {
+	return cloudvolumes.ListOpts{
+		AvailabilityZone:    d.Get("availability_zone").(string),
+		EnterpriseProjectID: d.Get("enterprise_project_id").(string),
+		Multiattach:         d.Get("shareable").(bool),
```

Comment:
`multiattach` is optional, it is better not to set a value when `shareable` is not set

Reply:
done